### ISSUE TITLE
[feature] CMakeToolchain + Emscripten PoC

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#7816]
Implement a proof of concept of using the new CMakeToolchain feature for building Emscripten/WebAsm projects:

- From Linux
- From Windows

Notes:
- Emscripten can be assumed installed
- Focused on creating/building 1 package, both local flow (conan install + native cmake) and ``conan create`` flow.
- Integration test.
- Test can be skipped for CI, but should work locally, annotate assumptions and installation details
- No env-var configuration at all
- All code must be private and local to the toolchains package
- New cross-build model with contexts must be used if necessary, not the old os_build, arch_build
- If enough time to implement reuse of a package, the ``cmake_find_package_multi`` generator must be used

**Comment by platc2** [comment]
Do I understand correctly this would include an example where I could use the following commands to build my project (Additional parameters are missing obviously)?

- `conan install`
- `cmake`
- `cmake --build`

I am looking for exactly this and I try to not include any conan macros in my `CMakeLists.txt`. I've got this working quite fine with `cmake_find_package_multi` and **no** Emscripten. Does PoC in this issue mean it would already be possible or are changes necessary for this?

**Comment by solvingj** [comment]
@platc2 yes, the current `cmake_find_package` generator now supports fully transparent (to the consuming `CMakeLists.txt`) delivery of Conan dependency and build variables to `CMake`.  In some cases, it depends a bit on the packages themselves, and whether they've been refactored with some recent changes (using `components` feature), but the most crucial ones I am aware of are `boost` and `openssl` and they both have been. 

So, here's example usage we used in a recent demo we did at C++ dublin meetup.  

The key remaining caveat to note is for the caller, and that is we must still tell `CMake` to look in the "current directory" for `.cmake` files first. 

Linux:  
```
        conan install ..
        cmake ..  -DCMAKE_BUILD_TYPE=Release  -DCMAKE_MODULE_PATH=$PWD
        cmake --build .   
```

Windows:
```
        conan install ..
        cmake ..  -DCMAKE_BUILD_TYPE=Release  -DCMAKE_MODULE_PATH=%CD:\=/%
        cmake --build . --config Release
```

**Comment by memsharded** [comment]
Implemented in https://github.com/conan-io/conan/pull/18432, for next Conan 2.18 release, feedback very welcome!
